### PR TITLE
fix error of RangeFrom in for-loop

### PR DESCRIPTION
**Pull request by bravomikekilo**
fix [issue-1542](https://github.com/rust-analyzer/rust-analyzer/issues/1542) @matklad

### CODE REVIEW

File: crates/ra_parser/src/grammar/expressions/atom.rs [hunk 1]
```diff
@@ -307,6 +313,16 @@ fn while_expr(p: &mut Parser, m: Option<Marker>) -> CompletedMarker {
 // fn foo() {
 //     for x in [] {};
 // }
+
+// test for_range_block
+// fn foo() {
+//    for x in 0 .. {
+//        break;
+//    }
```
Comment (matklad) [inline crates/ra_parser/src/grammar/expressions/atom.rs]: Let's move this test to `if r.forbid_structs` condition...

File: crates/ra_parser/src/grammar/expressions/atom.rs [hunk 2]
```diff
@@ -307,6 +313,16 @@ fn while_expr(p: &mut Parser, m: Option<Marker>) -> CompletedMarker {
...
+//    for _x in 0 .. (0 .. {1 + 2}).sum::<u32>() {
```
Comment (matklad) [inline crates/ra_parser/src/grammar/expressions/atom.rs]: ... and this test to test data, as it doesn't test a single isolated bit of a grammar and is rather a "stress test".

Comment (bravomikekilo) [reply crates/ra_parser/src/grammar/expressions/atom.rs]: ok

**Comment by bravomikekilo** [comment]
test position fixed @matklad

**Comment by matklad** [comment]
bors r+

Thanks @bravomikekilo !

**Comment by bravomikekilo** [comment]
github bors seems get some problem. @matklad

**Pull request merged**